WalletWasabi.Gui/Controls/LockScreen/SlideLock.xaml.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using Avalonia.Media;
+using Avalonia.Input;
+using Avalonia.Controls;
+using Avalonia.Markup.Xaml;
+using Avalonia.Controls.Primitives;
+using ReactiveUI;
+using Avalonia;
+
+namespace WalletWasabi.Gui.Controls.LockScreen
+{
+    internal class SlideLock : LockScreenBase
+    {
+        private Grid Shade;
+        private Thumb DragThumb;
+        private TranslateTransform TargetTransform;
+        private bool UserDragInProgress, StopSpring;
+
+        private const double ThresholdPercent = 1 / 6d;
+        private double RealThreshold;
+        private const double Stiffness = 0.12d;
+
+        private double _offset = 0;
+        private double Offset
+        {
+            get => _offset;
+            set => OnOffsetChanged(value);
+        }
+
+        public SlideLock() : base()
+        {
+            InitializeComponent();
+
+            this.Shade = this.FindControl<Grid>("Shade");
+            this.DragThumb = this.FindControl<Thumb>("PART_DragThumb");
+
+            TargetTransform = new TranslateTransform();
+
+            DragThumb.DragCompleted += OnDragCompleted;
```

Comment:
Don't subscribe to events on the GUI project. Make them observable with RX and make sure you're unsubscribing with disposewith. Sth like Observable.CreateFromEvent.